### PR TITLE
BIGTOP-4160: Auto switch docker compose yaml with host's cgroup version

### PR DESCRIPTION
**Pull request by yamasakisua**
### Description of PR
Change docker-compose.yml when user Debian container.
Keep "-F" option for flexibility, just in case.

### How was this patch tested?

```
$./docker-hadoop.sh -dcp -C config_ubuntu-22.04.yaml -c 3
...
Notice: /Stage[main]/Hadoop::Nodemanager/File[/data/2/yarn]/group: group changed 'root' to 'yarn' 
Notice: /Stage[main]/Hadoop::Nodemanager/Service[hadoop-yarn-nodemanager]/ensure: ensure changed 'stopped' to 'running'  
Notice: Applied catalog in 275.07 seconds
```

### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [x] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

**Comment by masatana** [comment]
Shouldn't we update the other distros?

**Comment by yamasakisua** [comment]
The docker-hadoop.sh can success when this patch is applied in my environment (WSL Ubuntu 22.04).
However, perhaps I was mistaken about the cause of failure.
It is caused by host environment, it should be switch with **host** cgroup environment. (my previous change is switch with **guest** environment)

I have tried several environments to see if the host is affected.

1. WSL Ubuntu 22.04
	1.  `docker info` shows `Cgroup Version: 1`
	2.  When use docker-compose.yml, provisioner fail.
	3.   When use docker-compose-cgroupv2.yml, provisioner success.
4. AWS(EC2) Ubuntu 22.04
	1.  `docker info` shows `Cgroup Version: 2`
	2.   When use docker-compose.yml, provisioner fail.
	3.   When use docker-compose-cgroupv2.yml, provisioner success.
5. AWS(EC2) Rocky 8
	1.  `docker info` shows `Cgroup Version: 1`
	2.  When use docker-compose.yml, provisioner success.

I'm now wondering if there was no difference in the success of the execution command between the 1 and 2 environments...

I guess WSL environment actually use cgroupv2 (not proof). 

I'm going to withdraw this pull request because I haven't come up with a way to switch it properly.

**Comment by yamasakisua** [comment]
This patch changes the docker-compose file depending on the host cgroup version

An explicit choice is also available, and in some cases, such as in a WSL environment, the “-F” option will continue to work.

**Comment by yamasakisua** [comment]
Changed the titles of PR as the content no longer matches the titles ( JIRA ticket too)

File: provisioner/docker/docker-hadoop.sh [hunk 1]
```diff
@@ -79,6 +79,23 @@ create() {
...
 
+    if [ -z ${distro+x} ]; then
+        distro=$(get-yaml-config distro)
+    fi
+
+    if [ "${enable_manual_docker_compose}" != true ]; then
+      docker_cgroup_version=`docker info | grep "Cgroup Version:" | awk '{print $3}'`
+      if [ ${docker_cgroup_version} == "1" ]; then
+        DOCKER_COMPOSE_CMD="${DOCKER_COMPOSE_CMD} -f docker-compose.yml"
+        log "Unknown cgroup version. cgroup v1 is used as default."
```
[the remark this answers is not in the document]
Comment (yamasakisua) [reply provisioner/docker/docker-hadoop.sh]: sorry, it's my mistake. I fixed it.

**Review by iwasakims — APPROVED** [review]
LGTM. worked for me on my Ubuntu 22.04 (cgroup v2) and Rocky Linux 8 (cgroup v1). Thanks, @yamasakisua.

**Pull request merged**